### ISSUE TITLE
Add end-to-end guide for configuring haskell-mode in a place where users will see it

### ISSUE THREAD

**Issue by tibbe** [haskell/haskell-mode#197]
One of the benefits of using Eclipse if you're a Java user is that everything just works out-of-the-box. Autocompletion, type information, etc. I've heard legends of haskell-mode being able to complete such feats (with the help of other pieces of software). Unfortunately I'm just a mere Haskell programmer and emacs-fu is beyond my abilities. Could those who have read and understood the sacred emacs lisp codes enlighten the rest of us with an end-to-end guide to setting up a complete emacs environment for Haskell development.

Bonus points for:
 * Supporting `cabal repl`.
 * Automating everything. Why should humans repeat step-by-step guides on wiki pages. We have machines for such tasks.

P.S. Someone shared this guide with me: http://lpaste.net/92858. Perhaps it can serve as a starting point.

**Comment by gracjan** [comment]
This issue report was last update in 2013. Is it still relevant to current code state?

**Comment by bgamari** [comment]
@chrisdone I know you have done a fair amount of work on Haskell mode. Perhaps you have something to add here?

**Comment by gracjan** [comment]
We are heading this direction, just look here: http://haskell.github.io/haskell-mode/manual/latest/

Technically manual has the following features:
- easy to edit with git history workflow
- can looks as good as css allows
- integrates well with Emacs standard Info browser
- has usual structure
- has animated gifs showing capabilities

There is this remaining part that somebody needs to sit down and write the content.

**Comment by ygale** [comment]
I want to upgrade to the latest haskell-mode.

I followed the link on this month's reddit post. Nothing there. But there was a link to "Home". Huh. Nothing there either. But there was a link to this issue. Aha, looks like I finally found it! I clicked on the link  which was the reason that this issue was closed. Well, not yet... but there is a link to "Getting Started". That must be it!

    "If you are reading this, you have most likely already managed to install Haskell Mode in one way or another."

Ugh. No no! I was reading it on the web page! Well... yes, I *did* install it. Years ago. Who remembers? I keep hearing about a bunch of different kinds of package managers, each with its own way of installing. Not a word, not a word in the Haskell Mode documentation. Or at least, not anything I could find even after a *lot* of searching.

Could we please, please have: an "end-to-end guide for configuring haskell-mode in a place where users will see it"?

In other words: please re-open this issue. Thanks.

**Comment by gracjan** [comment]
Okay :)

For now the documentation is a bit scattered. Starting from zero is described in readme here:

https://github.com/haskell/haskell-mode/blob/master/README.md

Note that it should work out of the box, there is not much you need to just start using it.

Advanced features are a bit different beasts though, so please come back for more information.

**Comment by gracjan** [comment]
It is here: http://haskell.github.io/haskell-mode/manual/latest/Installation.html#Installation
